python-package/lightgbm/sklearn.py

```diff
@@ -720,9 +720,8 @@ def predict(self, X, raw_score=False, start_iteration=0, num_iteration=None,
         n_features = X.shape[1]
         if self._n_features != n_features:
             raise ValueError("Number of features of the model must "
-                             "match the input. Model n_features_ is %s and "
-                             "input n_features is %s "
-                             % (self._n_features, n_features))
+                             f"match the input. Model n_features_ is {self._n_features} and "
+                             f"input n_features is {n_features} ")
         return self._Booster.predict(X, raw_score=raw_score, start_iteration=start_iteration, num_iteration=num_iteration,
                                      pred_leaf=pred_leaf, pred_contrib=pred_contrib, **kwargs)
 
```

Comment:
Please update the following lines within this PR as well
https://github.com/microsoft/LightGBM/blob/211ef7878f0c8a72757fd2b0050029908bc3091a/python-package/lightgbm/sklearn.py#L535-L536
https://github.com/microsoft/LightGBM/blob/211ef7878f0c8a72757fd2b0050029908bc3091a/python-package/lightgbm/sklearn.py#L997-L998 (part about `' ' * x`)
https://github.com/microsoft/LightGBM/blob/211ef7878f0c8a72757fd2b0050029908bc3091a/python-package/lightgbm/sklearn.py#L655

Reply:
@StrikerRUS do I only change the `' '*x` part or do I also modify the concatenation taking place there into an f-string as below?

<pre>
space = " "
    fit.__doc__ = (f"{_before_early_stop}" \
                  f"eval_at : iterable of int, optional (default=(1, 2, 3, 4, 5))\n" \
                  f"{space*12} The evaluation positions of the specified metric.\n" \
                  f"{space*8} {_early_stop} {_after_early_stop}")
</pre>

Reply:
Originally I thought about only `' '*x` part, but just saw that @jameslamb would prefer modifying concatenation as well: https://github.com/microsoft/LightGBM/pull/4144#discussion_r610993091. So, please modify it with concatenation.

BTW, there are some other places where concatenation can be eliminated with f-strings:
https://github.com/microsoft/LightGBM/blob/211ef7878f0c8a72757fd2b0050029908bc3091a/python-package/lightgbm/sklearn.py#L705-L711
https://github.com/microsoft/LightGBM/blob/211ef7878f0c8a72757fd2b0050029908bc3091a/python-package/lightgbm/sklearn.py#L825-L831
https://github.com/microsoft/LightGBM/blob/211ef7878f0c8a72757fd2b0050029908bc3091a/python-package/lightgbm/sklearn.py#L898-L902
https://github.com/microsoft/LightGBM/blob/211ef7878f0c8a72757fd2b0050029908bc3091a/python-package/lightgbm/sklearn.py#L990-L992
But I don't think we should replace concatenation with f-strings there because concatenation is more readable in those cases in my opinion. @jameslamb WDYT?


Also, please check the following comment about padding https://github.com/microsoft/LightGBM/pull/4144#discussion_r610994830.

Reply:
> But I don't think we should replace concatenation with f-strings there because concatenation is more readable in those cases in my opinion

Yes I agree with you.

Reply:
So I infer that no concatenations be migrated then.

Reply:
yes, cases like the ones @StrikerRUS mentioned (where `.format()` was used to replace a shared template like `_lgbmmodel_doc_fit` or where there are just variables being added to each other but not string literals) do not need to be changed for this pull request